### ISSUE TITLE
JSON widgets can't deal with extra / missing limbs

### ISSUE THREAD

**Issue by Venera3** [CleverRaven/Cataclysm-DDA#70860]
### Is your feature request related to a problem? Please describe.

The JSON sidebar can't handle custom bodyparts properly:
 - Missing (and not substituted) limbs are shown as 0HP while still taking up the space
 - Newly added limbs don't show up in the sidebar

### Solution you would like.

Populate the sidebar to show the character's current limbs (and current limbs only).

### Additional context

As it stands the sidebar widgets are hand-defined with an extra substitution step.

**Comment by GuardianDll** [comment]
That is a fundamental issue, since all widgets show information about limbs, always assuming character has 6 limbs, and occupying 6 rows to show the limb info
Additionally, we should think about all UI that uses ascii depiction of the humanoid - i don't think it makes any sense to show said human, if your legs are replaced with snail foot, and your arms are 8 tentacles now

**Comment by Venera3** [comment]
Redoing the bodygraphs is a separate problem, but they are very much secondary in my eyes - if your pwetty wings have ascii art for them is not really relevant as long as you can see at the first glance if they are too hurt to carry you down that cliff.

**Comment by Inglonias** [comment]
How much is known about extra limbs at compile time? Could we theoretically mark new limb health widgets for, say, bird wings in json without breaking things and throwing errors?

**Comment by Venera3** [comment]
Limbs live in data, so they are read on load. 

Flipping bp indicators to invisible when you don't have the part (and every limb needing its own JSON widget definition) would be an acceptable solution, even if it's harder to maintain.

**Comment by GuardianDll** [comment]
I think it even can be handled by json widget directly, as some condition "u_has_limb" or so
The issue is when your 6 rows with limbs, which are the base for the widget bar, should transform into 7 rows. Or 10 rows. Or 16. Or 3. Or even worse, since rows can be either empty or not, but can't collapse, without additional code in widgets it would look like "we have 40 rows to handle all possible limbs you may have, and 34 of them are pitch black with no info because you don't have any additional limbs"

**Comment by Inglonias** [comment]
A bandaid solution could be to have variants of the limb health widget for various mutation paths and just turn them off by default, allowing someone to use the sidebar config menu to turn them on as needed.

This is not ideal depending on how many paths we plan to have.

**Comment by GuardianDll** [comment]
No, as i explained in the previous message, this bandaid won't fix anything

**Comment by Inglonias** [comment]
Being able to show/hide various widgets depending on your situation would be useful for more than this. Some older sidebars replaced various elements when you were driving, which is behavior that I was not able to replicate when we transitioned sidebars to json

**Comment by GuardianDll** [comment]
there is a way to do it, again, it's just i think it needs to be updated, in some way
![GIF 12-01-2024 22-44-05](https://github.com/CleverRaven/Cataclysm-DDA/assets/67688115/61c7a17d-bb6b-415b-9c29-07eed5d5f1fd)
